File: backend/server_njs/controllers/loginController.js
require('dotenv').config();
const userModel = require('../models/userModel');
const jwt = require('jsonwebtoken');

class LoginController {
    constructor() { }

    async usersLogin(req, res) {
        try {
            const userEmail = req.params.email;
            const userPassword = req.params.password;
            const user = new userModel(null, null, null, userEmail, userPassword, null, null);
            const userByEmail = await user.getByEmail();
            if (!userByEmail) {
                res.status(501).json({ message: 'Account with that email not exist' });
            } else {
                if (userByEmail.Psw !== userPassword) {
                    res.status(501).json({ message: 'Incorrect password' });
                } else {
                    const token = jwt.sign({ userEmail, userPassword }, process.env.AUTH_KEY, { expiresIn: '1h' });
                    res.status(200).json({ token: token, message: 'Successful request' });
                }
            }
        } catch (err) {
            console.error(err);
            res.status(500).json({ message: 'Internal Server Error' });
        }
    }

    async passwordSend(req, res) {
        try {
            res.status(200).json({ message: 'Succesful request' });
        } catch (err) {
            console.error(err);
            res.status(500).json({ message: 'Internal Server Error' });
        }
    }

    async passwordUpdate(req, res) {
        try {
            res.status(200).json({ message: 'Succesful request' });
        } catch (err) {
            console.error(err);
            res.status(500).json({ message: 'Internal Server Error' });
        }
    }
}

module.exports = new LoginController();

File: backend/server_py/controllers/loginController.py
from flask import Blueprint, jsonify, request
from controllers.extServices.authMiddleware import authenticate_token, generate_token
from models.userModel import userModel
from typing import Optional

login_route = Blueprint('login_route', __name__)

@login_route.route('/<email>/<password>', methods=['GET'])
def usersLogin(email, password):
    try:
        user = userModel(None, None, None, email, password, None, None)
        user_by_email: Optional[userModel] = user.get_by_email()
        
        if not user_by_email:
            return jsonify({ 'message': 'Account with that email not exist' }), 501
        else:
            if not user.compare_hash(user_by_email['Psw']):
                return jsonify({ 'message': 'Incorrect password' }), 501
            else:
                token = generate_token(email)
                return jsonify({ 'id_User': user_by_email['Id'], 'token': token, 'message': 'Successful request' }), 200
            
    except Exception as e:
        print(e)
        return jsonify({ 'message': 'Internal Server Error' }), 500


@login_route.route('/passwordChange', methods=['POST'])
def passwordSend():
    try:
        return jsonify({ 'message': 'Succesful request' }), 200
    except Exception as e:
        print(e)
        return jsonify({ 'message': 'Internal Server Error' }), 500


@login_route.route('/passwordChange/update', methods=['POST'])
@authenticate_token
def passwordUpdate():
    try:
        return jsonify({ 'message': 'Succesful request' }), 200
    except Exception as e:
        print(e)
        return jsonify({ 'message': 'Internal Server Error' }), 500